packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -50,7 +50,7 @@ export abstract class PolicyPrincipal {
   /**
    * When this Principal is used in an AssumeRole policy, the action to use.
    */
-  public assumeRoleAction: string = 'sts:AssumeRole';
+  public assumeRoleAction: string | Array<string> = 'sts:AssumeRole';
```

Comment:
We don't expose type unions at the L2 level, so we can't accept this.

Reply:
How might this be achieved otherwise?  Any suggestions would be helpful.